src/nvim/event/process.c

```diff
@@ -112,11 +113,22 @@ bool process_spawn(Process *proc) FUNC_ATTR_NONNULL_ALL
 
 void process_teardown(Loop *loop) FUNC_ATTR_NONNULL_ALL
 {
+  process_is_tearing_down = true;
   kl_iter(WatcherPtr, loop->children, current) {
     Process *proc = (*current)->data;
-    uv_kill(proc->pid, SIGTERM);
-    proc->term_sent = true;
-    process_stop(proc);
+    if (proc->detach) {
+      // close handles to process without killing it.
+      CREATE_EVENT(loop->events, process_close_handles, 1, proc);
+    } else {
+      if (proc->type == kProcessTypeUv) {
+        uv_kill(proc->pid, SIGTERM);
+        proc->term_sent = true;
+        process_stop(proc);
+      } else { // kProcessTypePty
+        process_close_streams(proc);
+        pty_process_close_master((PtyProcess *)proc);
```

Comment:
That would mean moving `pty_process_teardown(loop)` to the top of this function?